### ISSUE TITLE
GenerateSetVersionMessage for 'Jenkins' does not set PreReleaseLabel correctly

### ISSUE THREAD

**Issue by kl4u5** [GitTools/GitVersion#1233]
I'm using GitVersionTask.3.6.5 and it works like a charm!

I've run into an issue when building via Jenkins.
It's as if the PreReleaseLabel does not get set according to settings in GitVersion.yml

My GitVersion.yml:
---
```
assembly-informational-format: '{NuGetVersion}.sha.{Sha}'
continuous-delivery-fallback-tag: a
mode: ContinuousDeployment
branches:
  master:
    tag: s
  releases?[/-]:
    tag: rc
  features?[/-]:
    tag: useBranchName
    increment: Minor
  (pull|pull\-requests|pr)[/-]:
    tag: pr
  hotfix(es)?[/-]:
    tag: rc
  support[/-]:
    tag: s
  dev(elop)?(ment)?$:
    tag: b
```
        
Output from jenkins when building the master branch:
---
```
12:40:00   INFO [06/08/17 12:40:00:07] Applicable build agent found: 'Jenkins'.
12:40:00   Executing GenerateSetVersionMessage for 'Jenkins'.
12:40:00   1.1.1-origin-master.11
12:40:00   Executing GenerateBuildLogOutput for 'Jenkins'.
12:40:00   GitVersion_Major=1
12:40:00   GitVersion_Minor=1
12:40:00   GitVersion_Patch=1
12:40:00   GitVersion_PreReleaseTag=origin-master.11
12:40:00   GitVersion_PreReleaseTagWithDash=-origin-master.11
12:40:00   GitVersion_PreReleaseLabel=origin-master
12:40:00   GitVersion_PreReleaseNumber=11
```

I would have expected GitVersion_PreReleaseLabel to be 's' but as you can see it is 'origin-master'

Any ideas?

**Comment by valakyr** [comment]
I think I've got the same issue.

I got Jenkins configured to build all feature branches via "origin/feature/**", which checks out all feature branches. But it doesn't checkout the feature branches locally, instead it uses the "origin/" prefix which confuses GitVersion into not recognizing the feature branch. GitVersion then handles the branch like any branch that isn't specifically configured and puts the branch name in to the version suffix.

You could fix this by prefixing your branch expressions with "(origin/)?", but that requires you to use GitVersion 4.0 if I'm not mistaken.

Another possibility could be to convince Jenkins to check out feature branches with local names, omitting the "origin/" prefix, but I don't think that is possible while using wildcards.

I'd definitely would like to hear about a better solution.

**Comment by valakyr** [comment]
Figured it out: Add the `Check out to specific local branch` behavior in Jenkins and set the "specific" branch to `**`. That will cause Jenkins to check out all branches locally, omitting the `origin/` prefix.

**Comment by abkljnAlmbrand** [comment]
Thanks valakyr. I'll give it a try and get back. Can't find the setting "Check out to specific local branch" - will get Sys Admin to have a try.

**Comment by asbjornu** [comment]
@valakyr, @abkljnAlmbrand: If you're able to confirm that this actually solves the problem, would one of you mind submitting a PR to [the Jenkins documentation](https://github.com/GitTools/GitVersion/blob/master/docs/build-server-support/build-server/jenkins.md) ([online here](http://gitversion.readthedocs.io/en/latest/build-server-support/build-server/jenkins/))? Thanks!

**Comment by abkljnAlmbrand** [comment]
@asbjornu - Yes I will. Still waiting for Sys Admi to get back from vacation - sometime next week.

**Comment by onuralp** [comment]
Hey @valakyr, @abkljnAlmbrand 
Do you have any better solution for this? I'm facing this issue and, wildcard local branch ( ** ) didn't work for me.

**Comment by valakyr** [comment]
I've never gotten around to trying it out, but I suppose the correct approach for building multiple branches in Jenkins is to use a "multi branch pipeline" project. So you might want to check that out.

**Comment by onuralp** [comment]
@valakyr 
 I've just figured it out.

GitVersion reads environment variables to get branch name. [See here](https://github.com/GitTools/GitVersion/blob/c2b0cbddb8d7933e92579038fedb3644c1b7005c/src/GitVersionCore/BuildServers/Jenkins.cs#L41) to understand how it does. 

Jenkins has a GIT_BRANCH environment variable which includes branch name with the 'origin' prefix. The problem is, GitVersion uses this env variable. 

But GitVersion first tries to get GIT_LOCAL_BRANCH, which is not set.

So I've added GIT_LOCAL_BRANCH as an environment variable and it has fixed.

```
pipeline {
    environment {
        GIT_LOCAL_BRANCH = "master"
        //GIT_LOCAL_BRANCH = "${env.ghprbSourceBranch}" // For PRs, I use ghprb plugin to build PRs. So it pass branch name with the ghprbSourceBranch param
    }
    //..
}```

Thank you anyway
